pymilvus/milvus_client/index.py

```diff
@@ -0,0 +1,56 @@
+class IndexParam:
+    def __init__(self, field_name: str, index_type: str, index_name: str, **kwargs):
+        self._field_name = field_name
+        self._index_name = index_name
+        self._index_type = index_type
+        self._kwargs = kwargs
+
+    @property
+    def field_name(self):
+        return self._field_name
+
+    @property
+    def index_name(self):
+        return self._indexname
+
+    @property
+    def index_type(self):
+        return self._index_type
+
+    def __iter__(self):
+        yield "field_name", self._field_name
+        yield "index_name", self._index_name
+        yield from self._kwargs.items()
+
+    def __str__(self):
+        return str(dict(self))
+
+    def __eq__(self, other: None):
+        if isinstance(other, self.__class__):
+            return dict(self) == dict(other)
+
+        if isinstance(other, dict):
+            return dict(self) == other
+        return False
+
+
+class IndexParams:
+    def __init__(self):
+        self._indexes = {}
+
+    def add_index(self, field_name: str, index_type: str = "", index_name: str = "", **kwargs):
```

Comment:
It's OK to add an index_type="" and index_name="" index, are we allowed to do this?

Reply:
yes, if index_type is "" ,it  actually means AutoIndex for vector field and defaultScalarIndexName for scalar field. Both are determined by server.

If `index_name` is empty, the server will translate it to a name that is the same as the name of field.